scripts/config-mock-build.sh

```diff
@@ -2,5 +2,7 @@
 BUILD_PATH=./build/test/mock-builds/port-$2-$(openssl rand -hex 12)
 mkdir -p $BUILD_PATH
 cp -r ./$1/* $BUILD_PATH
-grep ":8001" $BUILD_PATH -lr | xargs sed -i.bak -e "s/\:8001/\:$2/g"
+grep "\[TEST_REPLACEABLE_MOCK_PORT\]" $BUILD_PATH -lr | xargs sed -i.bak -e "s/\[TEST_REPLACEABLE_MOCK_PORT\]/$2/g"
+# Replace export const MOCK_PORT line with the provided mock port
+grep "export const MOCK_PORT = '\[TEST_REPLACEABLE_MOCK_PORT\]';" $BUILD_PATH -lr | xargs sed -i.bak -e "s/export const MOCK_PORT = '\[TEST_REPLACEABLE_MOCK_PORT\]';/export const MOCK_PORT = '$2';/"
```

Comment:
does second `xargs sed ...` replace anything?

as first `xargs sed ...` at line 5 finds all occurrences of `[TEST_REPLACEABLE_MOCK_PORT]` and replaces them with provided port address.

then `grep "export const MOCK_PORT = '\[TEST_REPLACEABLE_MOCK_PORT\]';"` probably won't find anything, as all `[TEST_REPLACEABLE_MOCK_PORT]` were already replaced.

Reply:
Yep!
You are right. Thank you.
I thought too much!